src/main/java/org/sonatype/nexus/ci/iq/IqApplication.groovy

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import hudson.model.Describable
+import hudson.model.Descriptor
+import hudson.util.FormValidation
+import jenkins.model.Jenkins
+import org.kohsuke.stapler.QueryParameter
+
+abstract class IqApplication
+    implements Describable<IqApplication>
+{
+  String applicationId
+  
+  protected IqApplication(final String applicationId) {
+    this.applicationId = applicationId
+  }
+
+  abstract String getName()
```

Comment:
I see we implemented this for the select and manual, but what is it used for?

Reply:
> I see we implemented this for the select and manual, but what is it used for?

Jenkins needs this class to be abstract to work properly.  To pass the code quality check I had to add an abstract method.

Reply:
> To pass the code quality check I had to add an abstract method.

Changed 83aa26a